repository/src/main/resources/db/changelog/db.changelog-rdm-2917.xml

```diff
@@ -0,0 +1,504 @@
+                <constraints nullable="true"/>
+            </column>
+            <column name="created_at" type="datetime" defaultValue="now()">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+
+        <addPrimaryKey columnNames="id"
+                       constraintName="pk_role"
+                       tableName="role"/>
+
+        <addForeignKeyConstraint baseColumnNames="case_type_id"
+                                 baseTableName="role"
+                                 constraintName="fk_role_case_type_id"
+                                 referencedColumnNames="id"
+                                 referencedTableName="case_type"/>
+
+        <addUniqueConstraint columnNames="case_type_id, reference"
+                             constraintName="unique_role_case_type_id_role_reference"
+                             tableName="role"/>
+
+        <!-- Add foreign key for case_type_id column linking to case_type table -->
+        <addForeignKeyConstraint baseColumnNames="case_type_id"
+                                 baseTableName="role"
+                                 constraintName="fk_role_case_type_id_case_type_id"
+                                 referencedColumnNames="id"
+                                 referencedTableName="case_type"/>
+
+        <!-- Change precision of reference, name and description columns in role table -->
+        <modifyDataType columnName="reference" newDataType="varchar(255)" tableName="role"/>
+        <modifyDataType columnName="name" newDataType="varchar(255)" tableName="role"/>
+        <modifyDataType columnName="description" newDataType="varchar(255)" tableName="role"/>
+
+        <!-- Add new security_classification, user_role_id and dtype fields to the role table -->
+        <addColumn tableName="role">
+            <column name="user_role_id" type="integer" >
+                <constraints nullable="true"/>
+            </column>
+            <column name="security_classification" type="security_classification" defaultValue="PUBLIC">
+                <constraints nullable="true"/>
+            </column>
+            <column name="dtype" type="varchar(10)">
+                <constraints nullable="false"/>
+            </column>
+        </addColumn>
+
+        <!-- copy table data to the new case_field_acl table -->
+        <sql>
+            insert into role (reference, name, description, case_type_id, security_classification, dtype)
+              select reference, name, description, case_type_id, 'PUBLIC','CASEROLE' as dtype from case_role;
+        </sql>
+        <!-- Add not null constraint for security_classification column in role table-->
+        <addNotNullConstraint columnDataType="security_classification"
+                              columnName="security_classification"
+                              defaultNullValue="PUBLIC"
+                              tableName="role"/>
```

Comment:
Given we are never in a situation where a row was given an empty value for this column, this constraint should be replaced by `nullable="false"` as part of the column definition when creating the table.